hyperbolic-rs/rudders/hmath.py

```diff
@@ -121,7 +121,7 @@ def hyp_distance(x, y, c):
     xy = tf.reduce_sum(x * y, axis=-1, keepdims=True)
     c1 = 1 - 2 * c * xy + c * y2
     c2 = 1 - c * x2
-    num = tf.sqrt(tf.square(c1) * x2 + tf.square(c2) * y2 - (2 * c1 * c2) * xy)
+    num = tf.sqrt(tf.maximum(tf.square(c1) * x2 + tf.square(c2) * y2 - (2 * c1 * c2) * xy, tf.zeros_like(x2)))
```

Comment:
Maybe it's good to make a test for this function/file. These core-math things are easy to get wrong and hard to know when you do.

Reply:
Done